app/main.py

```diff
@@ -1,8 +1,26 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    people_list = [
+        Person(person["name"], person["age"])
+        for person in people
+    ]
+
+    for person in people:
```

Comment:
Can you do all this in one iteration to optimize performance of your code?

Reply:
I'm not sure it's a good idea since the input is a `list` of `dict`, and a `spouse` attribute requires an object. To add a `person_object` to the result list, it's necessary to check if the `person_object` is already created (as a spouse for a previous object). Then, check if the `spouse_object` is already in `Person.people`, and if not, iterate further through the `people` list to find `age` to create it. This would decrease the readability of the code significantly with a slight performance boost. Should I still do it?